io/TextWriter.cpp

```diff
@@ -117,10 +117,10 @@ void TextWriter::ready(PointTableRef table)
     // Yes, this isn't efficient when, but it's simple.
     if (m_dimOrder.empty() || m_writeAllDims)
     {
-        Dimension::IdList all = table.layout()->dims();
-        for (auto di = all.begin(); di != all.end(); ++di)
-            if (!Utils::contains(m_dims, *di))
-                m_dims.push_back(*di);
+        for (const auto d : table.layout()->dims()) {
```

Comment:
`const auto d : table.layout()->dims()`
should read
`const auto& d : table.layout()->dims()`

Generally, I'd suggest careful review when replacing iterator-based loops to range-based loops to avoid unnecessary copying of potentially heavy objects.